### PR TITLE
first pass

### PR DESCRIPTION
**Pull request by orian34**

### CODE REVIEW

**Comment by enumag** [comment]
Hmm seeing these conditions with voice swapping I think I'd like to handle those cases differently. I'll try to do something about it.

**Comment by enumag** [comment]
@orian34 Done. Can you review my commit? What do you think?

**Comment by orian34** [comment]
Works fine, I was unsure about changing the text order, but if it's ok, it's probably better how you did it.

File: Update/_meak_002.txt [hunk 1]
```diff
@@ -1490,7 +1467,9 @@ void main()
 		   NULL, " Definitely.\"", Line_Normal);
 	ClearMessage();
 	DrawSceneWithMask("black", "maskdown", 1, 0, 300 );
-	DrawScene("background/ryuuketu1", 400 );
+	FadeOutBGM( 1, 300, FALSE );
+	DrawSceneWithMask("background/ryuuketu1", "maskaa", 1, 0, 400 );
```
Comment (orian34) [inline Update/_meak_002.txt]: *background car noises*

File: Update/_meak_003.txt [hunk 2]
```diff
@@ -68,8 +63,7 @@ void main()
 	OutputLine(NULL, "祟りでも陰謀でも、とにかく絶対裏に何か因果関係がある。」",
 		   NULL, " Whether it's a conspiracy or a curse, there has to be.\"", Line_Normal);
 	ClearMessage();
-	DrawScene("black", 400 );
-	PlayBGM( 1, "Mati", 120, 0 );
```
Comment (orian34) [inline Update/_meak_003.txt]: *background car noises intensifies*

File: Update/_meak_003.txt [hunk 3]
```diff
@@ -490,8 +484,9 @@ void main()
 	OutputLine(NULL, "…ちょっと驚いた。",
 		   NULL, " ...I was taken by surprise.", Line_Normal);
 	ClearMessage();
-	PlayBGM( 0, "higurashi", 120, 0 );
```
Comment (orian34) [inline Update/_meak_003.txt]: *menacing chirping intensifies*

File: Update/_meak_003.txt [hunk 4]
```diff
@@ -1648,8 +1645,8 @@ void main()
 	DrawScene("background/sora2", 1000 );
 
 	//VoiceMatching
-	if(GetGlobalFlag(GCensor) >= 2){ModCallScriptSection("_zmeak_003_vm0x_n01","dialog001");}
-	if(GetGlobalFlag(GCensor) <= 1){ModCallScriptSection("_zmeak_003_vm00_n01","dialog001");}
+	if(GetGlobalFlag(GCensor) >= 2){ModCallScriptSection("_zmeak_003_vm0x_n01","dialog002");}
```
Comment (orian34) [inline Update/_meak_003.txt]: *sentence intensifies*

File: Update/_meak_005.txt [hunk 5]
```diff
@@ -7,7 +7,8 @@ void main()
 	FadeOutBGM( 3, 0, FALSE );
 	ClearMessage();
 	DrawScene("white", 400 );
-	PlayBGM( 1, "Mati", 120, 0 );
```
Comment (orian34) [inline Update/_meak_005.txt]: *CAR NOISES INTENSIFIES*

File: Update/_meak_005.txt [hunk 6]
```diff
@@ -503,6 +516,9 @@ void main()
 		   NULL, " ...Whoa... ooh...\"", GetGlobalFlag(GLinemodeSp));
 	if (GetGlobalFlag(GADVMode)) { ClearMessage(); } else { OutputLineAll(NULL, "\n", Line_ContinueAfterTyping); }
 
+	FadeOutBGM( 0, 300, FALSE );
+	PlayBGM( 1, "t1", 128, 0 );
```
Comment (orian34) [inline Update/_meak_005.txt]: hau

File: Update/_meak_005.txt [hunk 7]
```diff
@@ -796,14 +813,10 @@ void main()
...
-	FadeOutBGM( 1, 0, FALSE );
-	FadeOutBGM( 2, 0, FALSE );
-	FadeOutBGM( 3, 0, FALSE );
-	ModDrawCharacter(19, 0, "transparent", "0", 0, 100, 0, FALSE, 0, 0, 0, 0, 0, 0, 0, 10, 0, TRUE );
+	FadeOutBGM( 1, 300, FALSE );
+	ModDrawCharacter(3, 7, "sprite/normal/sato1_komaru_", "0", 160, 0, 0, FALSE, 0, 0, 0, 0, 0, 0, 0, 20, 0, FALSE );
 	DrawSceneWithMask("background/tennai1", "maskleft", 1, 0, 300 );
-	ModDrawCharacter(3, 7, "sprite/normal/sato1_komaru_", "0", 160, 0, 0, FALSE, 0, 0, 0, 0, 0, 0, 0, 20, 200, TRUE );
-	PlayBGM( 0, "z2_hig2", 120, 0 );
+	PlayBGM( 0, "msys03", 120, 0 );
```
Comment (orian34) [inline Update/_meak_005.txt]: fun times ahead

File: Update/_meak_005.txt [hunk 8]
```diff
@@ -1580,23 +1586,27 @@ void main()
...
+	if (GetGlobalFlag(GADVMode)) { OutputLine("<color=#ffdc6f>悟史</color>", NULL, "<color=#ffdc6f>Satoshi</color>", NULL, Line_ContinueAfterTyping); }
 	ModPlayVoiceLS(4, 7, "ps3/s06/07/151200112", 128, TRUE);
 	OutputLine(NULL, "本当に助かったから。」",
 		   NULL, " That was sweet of you.\"", GetGlobalFlag(GLinemodeSp));
 	if (GetGlobalFlag(GADVMode)) { ClearMessage(); } else { OutputLineAll(NULL, "\n", Line_ContinueAfterTyping); }
-	PlaySE( 0, "s_boyoon", 128, 64 );
 
+	PlaySE( 0, "s_boyoon", 128, 64 );
+	FadeOutBGM( 0, 300, FALSE );
+	PlayBGM( 1, "t1", 128, 0 );
```
Comment (orian34) [inline Update/_meak_005.txt]: Hau!

File: Update/_meak_006.txt [hunk 9]
```diff
@@ -422,11 +420,19 @@ void main()
...
 		   NULL, " So...", Line_Continue);
-	PlaySE( 0, "s_boyoon", 128, 64 );
+	Wait( 1000 );
+	OutputLine(NULL, "…きゃふ！",
+		   NULL, " ...Whoa!", GetGlobalFlag(GLinemodeSp));
+	if (GetGlobalFlag(GADVMode)) { ClearMessage(); } else { OutputLineAll(NULL, "\n", Line_ContinueAfterTyping); }
+
+	DisableWindow();
+	FadeOutBGM( 0, 300, FALSE );
+	PlayBGM( 1, "t1", 128, 0 );
```
Comment (orian34) [inline Update/_meak_006.txt]: Hau~!

File: Update/_meak_006.txt [hunk 10]
```diff
@@ -1609,6 +1638,7 @@ void main()
 		   NULL, "\".....................\"", GetGlobalFlag(GLinemodeSp));
 	if (GetGlobalFlag(GADVMode)) { ClearMessage(); } else { OutputLineAll(NULL, "\n", Line_ContinueAfterTyping); }
 
+	PlayBGM( 0, "semi_r", 120, 0 );
```
Comment (orian34) [inline Update/_meak_006.txt]: *Erase his problems*

File: Update/_meak_016_2.txt [hunk 11]
```diff
@@ -2322,8 +2319,12 @@ void main()
 	if (GetGlobalFlag(GADVMode)) { OutputLine("<color=#f6d9a8>鷹野</color>", NULL, "<color=#f6d9a8>Takano</color>", NULL, Line_ContinueAfterTyping); }
 	ModPlayVoiceLS(4, 9, "ps3/s06/09/150900152", 128, TRUE);
 	OutputLine(NULL, "「…カクモ無惨且ツ残虐非道ヲ………。」",
-		   NULL, "\"...Never before has there been a corpse treated with such cruelty and inhumanity...\"", Line_Continue);
+		   NULL, "\"...Never before has there been a corpse treated with such cruelty and inhumanity...\"", GetGlobalFlag(GLinemodeSp));
+	if (GetGlobalFlag(GADVMode)) { ClearMessage(); } else { OutputLineAll(NULL, "\n", Line_ContinueAfterTyping); }
+	
 	SetSpeedOfMessage( TRUE, 128 );
+	
+	if (GetGlobalFlag(GADVMode)) { OutputLineAll("", NULL, Line_Continue); }
```
Comment (enumag) [inline Update/_meak_016_2.txt]: what is this good for?

File: Update/_meak_020.txt [hunk 12]
```diff
@@ -4504,15 +4507,17 @@ void main()
...
 		   NULL, " the kidnapping case of", Line_ContinueAfterTyping);
 	SetValidityOfInput( FALSE );
 	Wait( 1000 );
 	SetValidityOfInput( TRUE );
-	ModPlayVoiceLS(4, 18, "ps3/s06/18/153300133", 128, TRUE);
 	OutputLine(NULL, "あれの誘拐事件の話は知ってるだろ…？」",
-		   NULL, " the grandson of the Minister of construction, right?\"", GetGlobalFlag(GLinemodeSp));
+		   NULL, " the grandson of the Minister of construction, right?\"", Line_Continue);
+	WaitToFinishVoicePlaying(4);
+	ModPlayVoiceLS(4, 18, "ps3/s06/18/153300133", 128, TRUE);
```
Comment (enumag) [inline Update/_meak_020.txt]: In cases like this put both `WaitToFinishVoicePlaying` and `ModPlayVoiceLS` before the `OutputLine`. Revert the `GetGlobalFlag(GLinemodeSp)` too.

That one case when I put it after was an edge-case and the solution is not ideal either. Don't do that.

File: Update/_meak_021_1.txt [hunk 13]
```diff
@@ -528,10 +528,12 @@ void main()
...
 	OutputLine(NULL, "「祭具殿を侵した罪が、どれほどの大罪か。",
-		   NULL, "\"Don't you know", Line_WaitForInput);
-	ModPlayVoiceLS(4, 6, "ps3/s06/06/150601998", 128, TRUE);
+		   NULL, "\"Don't you know", Line_Continue);
 	OutputLine(NULL, "理解してないわけじゃないんでしょ？！」",
-		   NULL, " how serious a crime trespassing in the ritual storehouse is?!\"", GetGlobalFlag(GLinemodeSp));
+		   NULL, " how serious a crime trespassing in the ritual storehouse is?!\"", Line_Continue);
+	WaitToFinishVoicePlaying(4);
+	ModPlayVoiceLS(4, 6, "ps3/s06/06/150601998", 128, TRUE);
+	OutputLineAll(NULL, "", GetGlobalFlag(GLinemodeSp));
```
Comment (enumag) [inline Update/_meak_021_1.txt]: same here

File: Update/_meak_ep_02.txt [hunk 14]
```diff
@@ -203,7 +205,7 @@ void main()
 	OutputLineAll(NULL, "", GetGlobalFlag(GLinemodeSp));
 	if (GetGlobalFlag(GADVMode)) { ClearMessage(); } else { OutputLineAll(NULL, "\n", Line_ContinueAfterTyping); }
 
-	if (GetGlobalFlag(GADVMode)) { OutputLine("<color=#5ec69a>詩音</color>", NULL, "<color=#5ec69a>Shion</color>", NULL, Line_ContinueAfterTyping); }
+	if (GetGlobalFlag(GADVMode)) { OutputLine("<color=#5ec69a>魅音</color>", NULL, "<color=#5ec69a>Mion</color>", NULL, Line_ContinueAfterTyping); }
```
Comment (enumag) [inline Update/_meak_ep_02.txt]: This doesn't seem right. The file is in Shion's voice directory.

Comment (enumag) [reply Update/_meak_ep_02.txt]: Oh actually the name change might be correct, but the voice file is most likely wrong.

File: Update/_meak_016_2.txt (hunk 11 above)
Comment (orian34) [reply Update/_meak_016_2.txt]: to cut between dialogue and text.

File: Update/_meak_020.txt (hunk 12 above)
Comment (orian34) [reply Update/_meak_020.txt]: I must've been tired to not notice this solution, heh. It's fixed and will be added with tomorrow's changes

File: Update/_meak_007.txt [hunk 15]
```diff
@@ -838,24 +842,35 @@ void main()
...
-	FadeOutBGM( 1, 0, FALSE );
-	ModDrawCharacter(19, 0, "transparent", "0", 0, 100, 0, FALSE, 0, 0, 0, 0, 0, 0, 0, 10, 0, TRUE );
+	
+	DisableWindow();
+	SetValidityOfInterface(FALSE);
+	SetValidityOfSaving(FALSE);
+	FadeOutBGM( 0, 300, FALSE );
+	Wait( 200 ):
+	PlaySE( 4, "wa_027", 128, 64 );
+	DrawBustshot(6, "scene/002a", -500, 160, -790, FALSE, 0, 0, 0, 0, 0, 0, 0, 26, 50, TRUE );
```
Comment (orian34) [inline Update/_meak_007.txt]: praise Oyashiro-sama

File: Update/_meak_008.txt [hunk 16]
```diff
@@ -678,7 +683,11 @@ void main()
 	OutputLine(NULL, "　わぁあぁぁあぁああぁぁあん！！！」",
 		   NULL, " Waaaaahhhhhhh!!!\"", Line_Normal);
 	ClearMessage();
-	DrawScene("background/aka1", 400 );
+	
+	FadeOutBGM( 0, 1000, FALSE );
+	FadeOutBGM( 2, 1000, FALSE );
+	DrawScene("background/aka1", 1000 );
+	PlayBGM( 1, "msys23", 110, 0 );
```
Comment (orian34) [inline Update/_meak_008.txt]: the dam broke

File: Update/_meak_008.txt [hunk 17]
```diff
@@ -1396,31 +1423,35 @@ void main()
 	OutputLine(NULL, "　鬼のような形相で、私と悟史くんを交互に睨みつけながら、私たちの間に割って入る。",
 		   NULL, "She stepped in, looking as fierce as a demon.", Line_Normal);
 	ClearMessage();
-	ModDrawCharacter(2, 2, "sprite/normal/re1b_okoru_b1_", "0", 160, 0, 0, FALSE, 0, 0, 0, 0, 0, 0, 0, 10, 0, FALSE );
-	ModDrawCharacter(1, 7, "sprite/normal/sato1_ikari_", "1", -160, 0, 0, FALSE, 0, 0, 0, 0, 0, 0, 0, 8, 0, FALSE );
+	
+	ModDrawCharacter(3, 2, "sprite/normal/re1b_okoru_b1_", "0", 160, 0, 0, FALSE, 0, 0, 0, 0, 0, 0, 0, 20, 0, FALSE );
+	ModDrawCharacter(2, 7, "sprite/normal/sato1_ikari_", "1", -160, 0, 0, FALSE, 0, 0, 0, 0, 0, 0, 0, 10, 0, FALSE );
 	DrawScene("background/gk1", 400 );
+	PlayBGM( 1, "msys17", 128, 0 );
```
Comment (orian34) [inline Update/_meak_008.txt]: if you want music, use your imagination.

File: Update/_meak_008.txt [hunk 18]
```diff
@@ -1456,9 +1487,10 @@ void main()
 	OutputLine(NULL, "　悟史くんが何も言わなかったから、…私もまた沈黙を守ることにした。",
 		   NULL, "Since Satoshi-kun didn't say anything, I kept quiet, too.", Line_Normal);
 	ClearMessage();
-	ModDrawCharacterWithFiltering(2, 2, "sprite/normal/re1b_warai_b1_", "2", "maskright", 1, 160, 0, FALSE, 0, 0, 0, 0, 0, 10, 0, FALSE );
-	ModDrawCharacterWithFiltering(0, 4, "sprite/normal/sa1a_sakebu_a1_", "1", "maskright", 1, 0, 0, FALSE, 0, 0, 0, 0, 0, 9, 0, FALSE );
-	DrawSceneWithMask("background/gk1", "maskright", 1, 0, 300 );
```
Comment (orian34) [inline Update/_meak_008.txt]: what are those layers

**Comment by orian34** [comment]
I'll stop writing comments I think.

**Pull request merged**